include/ensmallen_bits/callbacks/traits.hpp

```diff
@@ -239,23 +239,11 @@ template<typename CallbackType,
          typename MatType>
 struct HasBeginOptimizationSignature
 {
-  const static bool hasBool =
+  const static bool value =
```

Comment:
Would it make sense to use `static constexpr bool` here, instead of plain `const` ?

Apart from the semantic info (clarity of intent), `constexpr` explicitly tells the compiler that this can be evaluated at compile time, instead of the compiler working this out for itself during optimisation.  This extra info should reduce compilation times somewhat.

Reply:
Ah, good point---fixed in e21a3c57. :+1: